packages/hawtio/src/plugins/shared/tree/node.ts

```diff
@@ -139,6 +146,11 @@ export class MBeanNode implements TreeViewDataItem {
 
     const remove: MBeanNode[] = this.children
     this.children = []
+
+    for (const r of remove) {
```

Comment:
Nitpicky style comment but I'd prefer to use `Array.forEach` or similar functional methods over `for` statement in most cases. We'd use `for (const x of xs)` only when we need to execute some async/await functions in loop but sequentially.
https://zellwk.com/blog/async-await-in-loops/

Reply:
I would as well but in this case, the for-loop must be synchronous in order that the `remove` children are returned correctly (with no parent). The `forEach` function cannot guarantee that and would require the unit test in node.test.ts to manage the asynchronousness.

Reply:
> The forEach function cannot guarantee that

I don't think so. JavaScript is single-thread model so it always runs synchronously unless you explicitly use setTimeout or async/await in the `forEach` callback. So it should guarantee the order. What I said in the first comment is about when you need to execute some async function or setTimeout inside a loop. Otherwise you should always be able to use `Array.forEach`.

Reply:
But for now let's mark it as resolved and move on.